api/src/client/ApiTaxFilingClient.ts

```diff
@@ -0,0 +1,167 @@
+import axios, { AxiosError } from "axios";
+import { flattenDeDupAndMapTaxFilings } from "../domain/tax-filings/taxIdHelper";
+import {
+  ApiTaxFilingClient,
+  TaxFilingLookupResponse,
+  TaxFilingOnboardingResponse,
+  TaxFilingResults,
+} from "../domain/types";
+import { LogWriterType } from "../libs/logWriter";
+
+type ApiConfig = {
+  apiKey: string;
+  baseUrl: string;
+};
+
+export const apiTaxFilingClient = (config: ApiConfig, logger: LogWriterType): ApiTaxFilingClient => {
```

Comment:
I almost thought that this was a violation of dependency inversion, but it's not quite. It's just written/named in a way that makes it look like it.

The idea behind dependency inversion is that nothing in our domain is specific to any particular implementation. Instead, it is an abstract descriptor of what occurs, in the interface type, and then when we implement it here, we provide a specific way of doing that abstract thing.

So in this case, it's incorrect to name something in our domain with the type name `ApiTaxFilingClient` because using the prefix `Api` is a specificity to _how_ it is happening - that is, that we're invoking an Api. instead, it should simply be `TaxFilingClient` and _this_ is the ApiTaxFilingClient - a _specific type_ of TaxFilingClient that _happens_ to invoke the NICUSA api.

It's just like how we would never say `BoomiClient` in our domain, because boomi is a specific way of doing it, but the abstract domain type is a `LicenseSearchClient`